### DISCUSSION TITLE
How to write an **async** babel plugin?

### DISCUSSION THREAD

**Discussion by NE-SmallTown** [babel/babel#11768] [Q&A] [answered]
Hi, is there any way to make a custom babel loader plugin async like https://webpack.js.org/api/loaders/#asynchronous-loaders . Thanks!

**Comment by NE-SmallTown** [comment]
Friendly ping~

**Comment by nicolo-ribaudo** [comment, marked as answer]
This is not possible yet.

If there are enough use cases, we can easily allow the `pre` and `post` plugin functions to be async (we already have the "infrastructure" to do so). However, visitors would likely never be asynchronous since we cannot run them in parallel and waiting one tick for every visitor method would be an enormous performance penality.

**Comment by RobertSandiford** [comment]
Is this still the situation?

I want to use import-meta-resolve to resolve and rewrite import locations - this is an async task

**Reply by liuxingbaoyu** [reply]
Yes, babel can't make any changes to this.
But you can make changes from the node side, for example
https://github.com/addaleax/synchronous-worker
https://github.com/un-ts/synckit

**Reply by RobertSandiford** [reply]
Cool, thanks for the tip
